src/core/server/saved_objects/service/lib/decorate_es_error.ts

```diff
@@ -69,6 +70,11 @@ export function decorateEsError(error: EsErrors) {
     if (match?.length > 0) {
       return SavedObjectsErrorHelpers.decorateIndexAliasNotFoundError(error, match[1]);
     }
+    // Throw EsUnavailable error if the 404 is not from elasticsearch
+    // Needed here to verify Product support for any non-ignored 404 responses from calls to ES
```

Comment:
Check against the product header and return the appropriate error from here to [catch](https://github.com/elastic/kibana/blob/master/src/core/server/saved_objects/service/lib/repository.ts#L1327-L1330) it in `update`.